src/Symfony/Component/Security/Http/AccessToken/QueryAccessTokenExtractor.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extracts a token from a query string parameter.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc6750#section-2.3
+ */
+final class QueryAccessTokenExtractor implements AccessTokenExtractorInterface
+{
+    public const PARAMETER = 'access_token';
+
+    public function __construct(private readonly string $parameter = self::PARAMETER)
+    {
+    }
+
+    public function extractAccessToken(Request $request): ?string
+    {
+        if (!$request->query->has($this->parameter)) {
+            return null;
+        }
```

Comment:
This block is useless since you check the type on line 38

Reply:
Correct